Peer/PeerThread.java

```diff
@@ -44,18 +45,31 @@ public void run()
             System.exit(1);
         }
 
-        //Closing statements
+        //Opening File
+        boolean openFile = false;
         try
         {
-            if(out != null)
-            {
-                out.close();
-            }
-            if(in != null)
+            File tempFile = new File(fileName);
+            if(Desktop.isDesktopSupported() && tempFile.exists())
```

Comment:
This will fail when inside a docker container. 

Remove anything which can't be run headless